### PR TITLE
executor: fix some bugs about new plan.

### PR DESCRIPTION
**Pull request by hanfei1991**
1. Let PointGet be able to process new executors.
2. fix bugs of dirty tables.

@shenli @coocood @zimulala @winoros @lamxTyler PTAL

### CODE REVIEW

File: executor/new_distsql.go [hunk 1]
```diff
@@ -43,6 +43,8 @@ type TableReaderExecutor struct {
 	dagPB     *tipb.DAGRequest
 	ctx       context.Context
 	schema    *expression.Schema
+	// columns are only required by us
```
Comment (zimulala) [inline executor/new_distsql.go]: What does `us` mean?

File: executor/executor_test.go [hunk 2]
```diff
@@ -1156,6 +1157,7 @@ func (s *testSuite) TestPointGet(c *C) {
 		c.Assert(ret, Equals, result)
 	}
 
+	plan.UseDAGPlanBuilder = false
```
Comment (shenli) [inline executor/executor_test.go]: Move this into defer.

File: executor/builder.go [hunk 3]
```diff
@@ -346,6 +346,13 @@ func (b *executorBuilder) buildUnionScanExec(v *plan.PhysicalUnionScan) Executor
 		us.desc = x.desc
 		us.dirty = getDirtyDB(b.ctx).getDirtyTable(x.table.Meta().ID)
 		us.conditions = v.Conditions
+		us.columns = x.Columns
```
Comment (shenli) [inline executor/builder.go]: Any test case over this?

Comment (hanfei1991) [inline executor/builder.go]: DirtyTable Test has been opened for new plan

**Comment by hanfei1991** [comment]
@shenli @zimulala PTAL

File: executor/adapter.go [hunk 4]
```diff
@@ -235,14 +235,21 @@ func IsPointGetWithPKOrUniqueKeyByAutoCommit(ctx context.Context, p plan.Plan) b
 	}
 
 	// get by index key
```
Comment (lamxTyler) [inline executor/adapter.go]: This can be deleted?

**Comment by shenli** [comment]
LGTM
@zimulala PTAL

**Comment by zimulala** [comment]
LGTM

**Pull request merged**